relays/bin-substrate/src/cli/relay_headers.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::cli::{CliChain, PrometheusParams, SourceConnectionParams, TargetConnectionParams, TargetSigningParams};
+use structopt::{clap::arg_enum, StructOpt};
+
+/// Start headers relayer process.
+#[derive(StructOpt)]
+pub struct RelayHeaders {
+	/// A bridge instance to relay headers for.
+	#[structopt(possible_values = &RelayHeadersBridge::variants(), case_insensitive = true)]
+	bridge: RelayHeadersBridge,
+	#[structopt(flatten)]
+	source: SourceConnectionParams,
+	#[structopt(flatten)]
+	target: TargetConnectionParams,
+	#[structopt(flatten)]
+	target_sign: TargetSigningParams,
+	#[structopt(flatten)]
+	prometheus_params: PrometheusParams,
+}
+
+// TODO [#851] Use kebab-case.
+arg_enum! {
+	#[derive(Debug)]
+	/// Headers relay bridge.
+	pub enum RelayHeadersBridge {
+		MillauToRialto,
+		RialtoToMillau,
+		WestendToMillau,
+	}
+}
+
+macro_rules! select_bridge {
+	($bridge: expr, $generic: tt) => {
```

Comment:
Same here w.r.t. to `generic`.

Reply:
It's actually not a closure, but a piece of arbitrary code that will use generic parameters `type Source =/ type Target = ` that is set in the particular branch of `match` statement.